api/tests/acceptance/application/organizations/organization-controller_test.js

```diff
@@ -1653,6 +1653,79 @@ describe('Acceptance | Application | organization-controller', function () {
     });
   });
 
+  describe('POST /api/admin/organizations/{id}/attach-target-profiles', function () {
+    let userId;
+    let organizationId;
+    let alreadyAttachedTargetProfileId;
+    let toAttachTargetProfileId;
+
+    beforeEach(async function () {
+      userId = databaseBuilder.factory.buildUser.withRole().id;
+      organizationId = databaseBuilder.factory.buildOrganization().id;
+      alreadyAttachedTargetProfileId = databaseBuilder.factory.buildTargetProfile().id;
+      toAttachTargetProfileId = databaseBuilder.factory.buildTargetProfile().id;
+      databaseBuilder.factory.buildTargetProfileShare({
+        organizationId,
+        targetProfileId: alreadyAttachedTargetProfileId,
+      });
+      await databaseBuilder.commit();
+    });
+
+    afterEach(async function () {
+      await knex('target-profile-shares').delete();
+    });
+
+    context('when target profiles to attach exists', function () {
+      it('should attach target profiles to organization', async function () {
+        // given
+        const options = {
+          method: 'POST',
+          url: `/api/admin/organizations/${organizationId}/attach-target-profiles`,
+          headers: { authorization: generateValidRequestAuthorizationHeader(userId) },
+          payload: {
+            'target-profile-ids': [alreadyAttachedTargetProfileId, toAttachTargetProfileId],
+          },
+        };
+
+        // when
+        const response = await server.inject(options);
+
+        // then
+        const attachedTargetProfileIds = await knex('target-profile-shares')
+          .pluck('targetProfileId')
+          .where({ organizationId })
+          .orderBy('targetProfileId', 'ASC');
+        expect(response.statusCode).to.equal(204);
+        expect(attachedTargetProfileIds).to.deepEqualArray([alreadyAttachedTargetProfileId, toAttachTargetProfileId]);
+      });
+    });
+
+    context('when a target profile does not exist', function () {
+      it('should return a 404 error without attaching no more target profile', async function () {
```

Comment:
**Suggestion**: "should return a 404 error without attaching **any** target profile"